### PR TITLE
DAOS-11418 container: rename redundancy container properties

### PR DESCRIPTION
**Pull request by knard-intel**
Respectively rename the rf and rf_lvl properties to rd_fac and rd_lvl.

Required-githooks: true
Signed-off-by: Cedric Koch-Hofer <cedric.koch-hofer@intel.com>

### CODE REVIEW

**Comment by knard-intel** [comment]
@phender could you tell me if I could run this PR with the daily regression tag to increase its coverage ?

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Comment by knard-intel** [comment]
@wangshilong added you as reviewers as I have updated the code in _src/client/pydaos/raw/daos_api.py_ tha you have pushed into https://github.com/daos-stack/daos/pull/10203

**Comment by wangshilong** [comment]
> @wangshilong added you as reviewers as I have updated the code in _src/client/pydaos/raw/daos_api.py_ tha you have pushed into your PR [DAOS-11375 pydaos: add rf_lvl for cont create api](https://github.com/daos-stack/daos/pull/10203)

Thanks, modifications looks fine, just one thing, Above PR use following tag to make sure it fixed original failure:
Test-tag: pr ec_disabled_rebuild_single test_query_properties

However, i noticed that this PR use features: container, not sure it will include above test cases?

**Comment by knard-intel** [comment]
> > @wangshilong added you as reviewers as I have updated the code in _src/client/pydaos/raw/daos_api.py_ tha you have pushed into your PR [DAOS-11375 pydaos: add rf_lvl for cont create api](https://github.com/daos-stack/daos/pull/10203)
> 
> Thanks, modifications looks fine, just one thing, Above PR use following tag to make sure it fixed original failure: Test-tag: pr ec_disabled_rebuild_single test_query_properties
> 
> However, i noticed that this PR use features: container, not sure it will include above test cases?

From what I have seen these tests are not included with the tag container: I am going to rebase the commit messages and relaunch the tests.

- [ ] Update commit messages with tags ec_disabled_rebuild_single test_query_properties

File: src/control/cmd/daos/property.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ type propHdlr struct {
 	readOnly bool
 }
 
+type propDeprHdlrMap map[string]string
+
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply src/control/cmd/daos/property.go:59]: - [ ] Remove useless type alias

**Comment by knard-intel** [comment]
Merge with latest tip of master have introduce some regression.
I am going to fix this.

File: src/include/daos/cont_props.h [hunk 2]
```diff
@@ -32,6 +32,10 @@
 #define DAOS_PROP_ENTRY_RP_PDA		"rp_pda"
 #define DAOS_PROP_ENTRY_GLOBAL_VERSION	"global_version"
 
+/** DAOS deprecated property entry names keeped for backward compatibility */
+#define DAOS_PROP_ENTRY_REDUN_FAC_OLD	"rf"
+#define DAOS_PROP_ENTRY_REDUN_LVL_OLD	"rf_lvl"
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply src/include/daos/cont_props.h:37]: It is also needed for serialization/deserialization which could be done from C code in src/common/prop.c.
Without these macros and code, some functional daos-vol tests where broken.

File: src/control/lib/ui/flags.go [hunk 3]
```diff
@@ -116,6 +117,14 @@ func (f *SetPropertiesFlag) SettableKeys(keys ...string) {
 	}
 }
 
+// SettableDeprecated Keys accepts a list of deprecated property keys that are settable.
+func (f *SetPropertiesFlag) SettableDeprecatedKeys(keys ...string) {
+	f.settableDeprecatedKeys = make(common.StringSet)
+	for _, key := range keys {
+		f.settableDeprecatedKeys.Add(key)
+	}
+}
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply src/control/lib/ui/flags.go:126]: I have to admit, that I did not remember exactly why I was not able to simply merge the two sets.
I am going to try to do it as you suggested and check if there is no issue to do that.

[the remark this answers is not in the document]
Comment (knard-intel) [reply src/control/lib/ui/flags.go]: Thanks to your remarks, I have refactored the code to put as much as possible management codes of deprecated keys in the base file src/control/lib/ui/flags.go: this is allowing to factorize much more code between dmg and daos command. However, I had to kept some management codes of deprecated keys in the daos cmd get_properties: this part of the code is not using the base function defined in the file src/control/lib/ui/flags.go.

**Comment by knard-intel** [comment]
There is only one test failing which is not related to this PR.
Indeed, this test also fail on master as illustrated by the following PR and its associated build:
* https://github.com/daos-stack/daos/pull/10396
* https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-10396/2/tests

**Comment by knard-intel** [comment]
> It's helpful for gatekeepers to see a ticket for the known failure. In this case, it appears to be [DAOS-11643](https://daosio.atlassian.net/browse/DAOS-11643).

Did not find the reference to this ticket in the test failure pages and thus I do not know how to find it.
Please, could you tell me how you find it.
Thanks for the ticket reference :)

**Comment by mjmac** [comment]
> > It's helpful for gatekeepers to see a ticket for the known failure. In this case, it appears to be [DAOS-11643](https://daosio.atlassian.net/browse/DAOS-11643).
> 
> Did not find the reference to this ticket in the test failure pages and thus I do not know how to find it. Please, could you tell me how you find it. Thanks for the ticket reference :)

The easiest way is just to search for the failing test. I just put `container/async.py` in the search field.

**Review by mjmac — APPROVED** [review]
Go changes look fine to me.

**Comment by wangshilong** [comment]
PR needs be rebased to resolve conflicts.

**Review by wangshilong — DISMISSED** [review]

**Comment by knard-intel** [comment]
The errors reported by the last build are known CI issues and thus should allow the forced landing:
* [daos_racer/simple_py](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10071/28/testReport/FTEST_launch/daos_racer-simple/Test_Hardware___Functional_Hardware_Large___07___daos_racer_simple_py) : JIRA Ticket [DAOS-11762](https://daosio.atlassian.net/browse/DAOS-11762)
* [daos_test/dfs.py](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10071/28/testReport/FTEST_launch/daos_test-dfs/Test_Hardware___Functional_Hardware_Large___09___daos_test_dfs_py/): JIRA ticket [DAOS-11763](https://daosio.atlassian.net/browse/DAOS-11763)
* [python bandit check](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10071/28/testReport/._src_tests_ftest_harness_advanced/py/Pre_build___Python_Bandit_check___blacklist/) : [Python Bandir Check](https://daosio.atlassian.net/browse/DAOS-11766)

**Review by mjmac — APPROVED** [review]
Go changes still look fine; no significant differences from the originally reviewed commits.

**Comment by knard-intel** [comment]
> Go changes still look fine; no significant differences from the originally reviewed commits.

Indeed, most of latest changes are just merge conflicts with master.

**Comment by wangshilong** [comment]
PR need be refreshed to resolve conflicts.

**Review by wangshilong — APPROVED** [review]

**Comment by knard-intel** [comment]
The error reported by the last build [#33](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10071/33/) is a known CI issue and thus should allow the Force Landing:
[daos_test/suite.py:DaosCoreTest.test_daos_pool - POOL14: pool connect access based on ACL](https://daosio.atlassian.net/browse/DAOS-10301)

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**